### DISCUSSION TITLE
Semster 4 Weekly Report 1

### DISCUSSION THREAD

**Discussion by BenjaminB109** [SE-TINF22B6/Underwatch#121] [Announcements]
# Semster 4 Weekly Report 1

## Summary of the current status?
As of now we have completed most major tasks in each of the subprojects (namely Game, Webpage, Database (WebService) and Design):
The Webpage has a working navigation, scoreboards where you can compare and filter scores as well as the foundational groundwork for the Wiki.

![Screenshot 2024-04-16 121138](https://github.com/SE-TINF22B6/Underwatch/assets/123720714/3058765a-f73d-4de1-8053-aab2a7e2c978)

The Database is in a state which can be deployed to a cloud provider (tho this hasn't been done yet). Also we use mock/sample data instead of real submitted scores at the moment.

In the Game we have implemented a third party api, the DadjokeAPI, which displays a joke on the start screen when you start the program.
You can now lose lives in the game and go game over, which completes the game loop.
We also implemented a real time lighting system which throws dynamic shadows based on the positions of objects in the game.

<img width="1440" alt="Bildschirmfoto 2024-04-16 um 11 55 24" src="https://github.com/SE-TINF22B6/Underwatch/assets/122724021/43a77930-46c3-4740-a0f1-472fb6215d02">
<img width="536" alt="Bildschirmfoto 2024-04-16 um 11 55 37" src="https://github.com/SE-TINF22B6/Underwatch/assets/122724021/7736de91-7936-46f5-a57f-e7d99d9a5a29">

## Which features have been done?

[TODO]

## Which features should be done in this Semester?

This semester we hope to complete the following features:

- Campaign Mode / Story Mode
- Content for the Game
- Pathfinding of enemies
- Diversity in Weapons
- Expand the Webpage to feature a Wiki with all the ingame lore

## Are there any milestones? When and what?

Currently there are only two milestones we have planned for:

- Completely overhaul the current documentation to fit the requirements put forwards by Mrs. Li (End of Semester)
- Complete the MVP of the entire system (as of now we don't upload scores and query them) (Sprint 8)

## Initial thought about the architecture of your project?
We initially thought about using an event-driven architecture for our game, as this makes the most sense for a game. Before we decide, however, we will wait for the upcoming lecture.

**Comment by LeonmAndreesen** [comment]
Good day, 

your looks really cool. Your state of development looks really good. You might be the most advanced. I really like your plan for the next steps.

**Comment by giomskii** [comment]
Hello team Underwatch,
your wiki page looks quite clean and compact with all the relevant information on the front. It was also a smart move, to include the DadJokeApi in your project since it's quite easy to implement (i assume), fullfills one of the minimum requirements of the project and is something you might expect in other games. We also look forward to your architecture choice, since it might differ from the majority of the groups. 
Keep up the good work! 👍🏼

**Comment by MargotGsb** [comment]
Dear Team Underwatch,
your Report looks really good. Your goals seem very achievable and your progress is truly amazing. I think you got a small mistake in the Section "Which features have been done?". There is only this "[TODO]. Was this on purpose? I am also excited to see your finished project and which architecture design you will choose for it!
Good Job! And Greetings from Team Plapy :D
